server/vertx/src/main/scala/sttp/tapir/server/vertx/VertxBodyListener.scala

```diff
@@ -0,0 +1,20 @@
+package sttp.tapir.server.vertx
+
+import io.vertx.ext.web.RoutingContext
+import sttp.monad.MonadError
+import sttp.tapir.server.interpreter.BodyListener
+
+import scala.util.{Success, Try}
+
+class VertxBodyListener[F[_]](implicit m: MonadError[F]) extends BodyListener[F, RoutingContext => Unit] {
+  override def onComplete(body: RoutingContext => Unit)(cb: Try[Unit] => F[Unit]): F[RoutingContext => Unit] = {
+    m.unit {
+      { (ctx: RoutingContext) =>
+        body {
+          ctx.addBodyEndHandler(_ => cb(Success(())))
```

Comment:
no way to intercept an exception?

Reply:
`addBodyEndHandler` callback will not be executed on exception, there's `addEndHandler` which will be executed but `will be called when the response is disposed or an exception has been encountered to allow consistent cleanup`, so it's not the moment we want to capture here right, since it has to be a body failure?

Reply:
can't we use both, one for success, another for error?